runtime/cudaq/platform/pasqal/PasqalServerHelper.cpp

```diff
@@ -52,50 +55,60 @@ RestHeaders PasqalServerHelper::getHeaders() {
 
 ServerJobPayload
 PasqalServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
-  ServerJobPayload ret;
-  std::vector<ServerMessage> &tasks = std::get<2>(ret);
+  std::vector<ServerMessage> tasks;
+
   for (auto &circuitCode : circuitCodes) {
-    // Construct the job message
-    ServerMessage taskRequest;
+    ServerMessage message;
+    message["name"] = circuitCode.name;
+    message["device"] = backendConfig.at("machine");
+    message["shots"] = shots;
 
-    taskRequest["name"] = circuitCode.name;
-    taskRequest["device"] = backendConfig.at("machine");
     auto action = nlohmann::json::parse(circuitCode.code);
-    taskRequest["action"] = action.dump();
-    taskRequest["shots"] = shots;
+    message["action"] = action.dump();
 
-    tasks.push_back(taskRequest);
+    tasks.push_back(message);
   }
 
   cudaq::info("Created job payload for Pasqal, targeting device {}",
               backendConfig.at("machine"));
   
   // Return a tuple containing the job path, headers, and the job message
-  return std::make_tuple(baseUrl + "/v1/cuda-q/job", getHeaders(), tasks);
+  return std::make_tuple(baseUrl + apiPath + "/cudaq/job", getHeaders(), tasks);
 }
 
 std::string PasqalServerHelper::extractJobId(ServerMessage &postResponse) {
-    return baseUrl + "";
+    return postResponse["id"].get<std::string>();
 }
 
 std::string PasqalServerHelper::constructGetJobPath(std::string &jobId) {
-  return baseUrl + "";
+  return baseUrl + apiPath + "/jobs/" + jobId + "/results_link";
 }
 
 std::string
 PasqalServerHelper::constructGetJobPath(ServerMessage &postResponse) {
-    return baseUrl + "";
+    return baseUrl + apiPath + "/jobs/" +
+      postResponse["id"].get<std::string>() + "/results_link";
 }
 
 bool PasqalServerHelper::jobIsDone(ServerMessage &getJobResponse) {
-    return false;
+  std::unordered_set<std::string>
+  terminals = {"PENDING", "RUNNING", "DONE", "ERROR", "CANCEL"};
+  
+  auto jobStatus = getJobResponse["status"].get<std::string>();
+  return terminals.find(jobStatus) != terminals.end();
 }
 
+// TODO: Implementing `processResults`.
```

Comment:
Since we'll be re-using the existing API, you can obtain the returned JSON many ways but it is e.g. here:

https://github.com/pasqal-io/pasqal-cloud/blob/dev/tests/fixtures/api/v1/batches/00000000-0000-0000-0000-000000000001/_.GET.json

We should use the `counter` under `full_results`.

Reply:
A simplified version is available. There is one point related to the index order that still needs to be checked later.